apitools/base/protorpclite/protojson.py

```diff
@@ -283,11 +283,19 @@ def __decode_dictionary(self, message_type, dictionary):
                 valid_value = [self.decode_field(field, item)
                                for item in value]
                 setattr(message, field.name, valid_value)
-            else:
-                # This is just for consistency with the old behavior.
-                if value == []:
-                    continue
+                continue
+            # This is just for consistency with the old behavior.
+            if value == []:
+                continue
+            try:
                 setattr(message, field.name, self.decode_field(field, value))
+            except messages.DecodeError:
+                # Save unknown enum values.
+                if not isinstance(field, messages.EnumField):
+                    raise
```

Comment:
When is this going to happen? Is raising the decode error sufficient in that situation?

Reply:
This happens in decode_field for misformatted datetime fields, base64 encoding errors, and message fields that can't be decoded recursively. I believe this is sufficient because those all reflect actual problems with the encoded message.